crates/bevy_ecs/src/world/component_constants.rs

```diff
@@ -16,24 +16,24 @@ pub const ON_REMOVE: ComponentId = ComponentId::new(3);
 
 /// Trigger emitted when a component is added to an entity. See [`crate::component::ComponentHooks::on_add`]
 /// for more information.
-#[derive(Event)]
+#[derive(Event, Debug)]
 #[cfg_attr(feature = "bevy_reflect", derive(Reflect))]
```

Comment:
```suggestion
#[cfg_attr(feature = "bevy_reflect", derive(Reflect))]
#[cfg_attr(feature = "bevy_reflect", reflect(Debug))]
```

This is a thing but I'm not sure if it's any useful for unit structs

Reply:
It does exist but you’re right that it doesn't really do anything special other than delegate `dyn Reflect`'s `Debug` to this one.

One possible difference though is that I think the reflection-based debugging uses the full type path rather than just the struct ident.

Reply:
Added